### ISSUE TITLE
Default value for sync-method flag breaks previous behaviour

### ISSUE THREAD

**Issue by paul-pop** [awslabs/ssosync#37]
**Describe the bug**
I noticed after upgrading to `v1.0.0-rc.9` from `v1.0.0-rc.8` that our SSO sync started failing with the following message:

```
time="2021-03-29T12:07:40Z" level=info msg="Syncing AWS users and groups from Google Workspace SAML Application"
time="2021-03-29T12:07:40Z" level=info msg=syncing sync_method=groups
time="2021-03-29T12:07:40Z" level=debug msg=get google groups query=
time="2021-03-29T12:07:40Z" level=fatal msg="googleapi: Error 400: Invalid Input: query, invalid"
```

The default value for the new `sync-method` flag is `groups` and I believe the empty query is not accepted by the Google API.

**To Reproduce**
Steps to reproduce the behaviour:
1. Run ssosync only with the minimum set of flags, for example:

```bash
ssosync \
          --debug \
          --access-token REDACTED \
          --endpoint REDACTED \
          --google-admin REDACTED \
          --google-credentials credentials.json
```

2. Check output

**Expected behavior**
The sync should happen with no errors.

**Additionaal context**
Even when setting the `sync-method` to `users_groups`, it yields the same result as the query to fetch groups is invalid.

**Comment by paul-pop** [comment]
Workaround for me was to set the `sync-method` flag to `users_groups`

**Comment by paul-pop** [comment]
I can confirm the issue is actually the invalid query. Once I pass in something like `--group-match 'email:*'` it behaves as it did before.

**Comment by joshuachong** [comment]
Hi @paul-pop
Apologies for the delayed response.
Part of #36 , it introduced the Query to have a more specific filter. Looks like this breaks previous inputs if `--group-match` or `--user-match` is not specified.

Are you good by adding the additional `--group-match`?

**Comment by paul-pop** [comment]
@joshuachong yes that works well for me, just flagging this for people using defaults

**Comment by cep21** [comment]
I'm getting the same error (above) even with the extra parameter.  Am I missing something?

```
+ go build .
+ ./ssosync -d --log-level debug -u X@X --group-match 'email:*'
INFO[0000] Syncing AWS users and groups from Google Workspace SAML Application
INFO[0000] syncing                                       sync_method=groups
DEBU[0000] get google groups                             query="email:*"
FATA[0000] googleapi: Error 400: Invalid Input, invalid
```

**Comment by jverhoeks** [comment]
There is no * possible with search. You need to add at least 1 character  like email:a*

**Comment by cep21** [comment]
Paul mentioned it working.  How do I get the old behavior of syncing all google groups?

**Comment by cep21** [comment]
@christiangda we want to sync both users and groups, and we want to match all groups, like the old behavior, but we cannot because of the issues mentioned in this thread.

> # in my case I have groups prefix AWS-SSO-<... my groups name ...>

This stipulation is not valid for us.

**Comment by christiangda** [comment]
@cep21 what I understand is that if you set

```
--sync-method users_groups
```

and you don't set this parameter `--group-match` you get some errors, isn't it?

**Comment by cep21** [comment]
> and you don't set this parameter --group-match you get some errors, isn't it?

That is correct.

**Comment by christiangda** [comment]
I'm working in a PR to fix this error, I mean when `--group-match` and `--user-match` are not used, also I'm documented better the use of the function.

**Comment by joshuachong** [comment]
Hi all, i've merged @christiangda 's PR #44 
Many thanks for your contribution on documentation updates as well.

**Comment by christiangda** [comment]
Hi guys, I finished the first version of the new alternative project to this one.

https://github.com/slashdevops/idp-scim-sync is a fresh implementation, it is not a `ssosync` fork, and the main idea is to keep a "state file" after the sync to avoid unnecessary requests to AWS SSO SCIM API.

this project is composed of different artefacts repository but the most convenient one is the `AWS Serverless Application` https://serverlessrepo.aws.amazon.com/applications/us-east-1/889836709304/idp-scim-sync
